### PR TITLE
unit test for macros in Custom Actions not addressed in PR-6206

### PR DESCRIPTION
**Pull request by thegreenfrog**
PR this unit tests covers:
https://github.com/caskdata/cdap/pull/6206

### CODE REVIEW

**Comment by thegreenfrog** [comment]
The Guild is Breen @knazir: http://builds.cask.co/browse/CDAP-DUT4572

File: cdap-app-templates/cdap-etl/cdap-data-pipeline/src/test/java/co/cask/cdap/datapipeline/DataPipelineTest.java [hunk 1]
```diff
@@ -27,6 +27,8 @@
 import co.cask.cdap.datapipeline.mock.NaiveBayesClassifier;
 import co.cask.cdap.datapipeline.mock.NaiveBayesTrainer;
 import co.cask.cdap.datapipeline.mock.SpamMessage;
+import co.cask.cdap.etl.api.batch.BatchSink;
+import co.cask.cdap.etl.api.batch.BatchSource;
```
Comment (knazir) [inline cdap-app-templates/cdap-etl/cdap-data-pipeline/src/test/java/co/cask/cdap/datapipeline/DataPipelineTest.java]: These two seem like unused imports

File: cdap-app-templates/cdap-etl/cdap-data-pipeline/src/test/java/co/cask/cdap/datapipeline/DataPipelineTest.java [hunk 2]
```diff
@@ -42,13 +44,16 @@
...
+import co.cask.cdap.etl.proto.Connection;
 import co.cask.cdap.etl.proto.Engine;
 import co.cask.cdap.etl.proto.v2.ETLBatchConfig;
 import co.cask.cdap.etl.proto.v2.ETLPlugin;
 import co.cask.cdap.etl.proto.v2.ETLStage;
 import co.cask.cdap.proto.Id;
 import co.cask.cdap.proto.ProgramRunStatus;
+import co.cask.cdap.proto.ProgramType;
 import co.cask.cdap.proto.RunRecord;
+import co.cask.cdap.proto.WorkflowNodeStateDetail;
```
Comment (knazir) [inline cdap-app-templates/cdap-etl/cdap-data-pipeline/src/test/java/co/cask/cdap/datapipeline/DataPipelineTest.java]: Also an unused import?

File: cdap-app-templates/cdap-etl/cdap-data-pipeline/src/test/java/co/cask/cdap/datapipeline/DataPipelineTest.java [hunk 3]
```diff
@@ -109,6 +114,41 @@ public void cleanupTest() throws Exception {
...
+      ETLStage action1 = new ETLStage("action1", MockAction.getPlugin("actionTable", "action1.row", "action1.column",
+                                                                               "${value}"));
+      ETLStage action2 = new ETLStage("action2", MockAction.getPlugin("actionTable", "action2.row", "action2.column",
+                                                                     "action2.value"));
+
+      ETLBatchConfig etlConfig = co.cask.cdap.etl.proto.v2.ETLBatchConfig.builder("* * * * *")
+        .addStage(action1)
+        .addStage(action2)
+        .addConnection(new Connection(action1.getName(), action2.getName()))
+        .setEngine(Engine.MAPREDUCE)
```
Comment (knazir) [inline cdap-app-templates/cdap-etl/cdap-data-pipeline/src/test/java/co/cask/cdap/datapipeline/DataPipelineTest.java]: May be good to have a spark test as well. Similar to `testSecureStoreMacroPipelines()`

**Comment by knazir** [comment]
Some minor comments and one about an additional test case @thegreenfrog

**Comment by thegreenfrog** [comment]
@knazir made edits to your comments.

**Comment by thegreenfrog** [comment]
Guild is Breen again: http://builds.cask.co/browse/CDAP-DUT4572-2

**Comment by albertshau** [comment]
lgtm

**Pull request merged**